source/develop/release-management/features/network/import-vm-net-changes.html.md

```diff
@@ -0,0 +1,164 @@
+</action>
+```
+
+Since the mappings are optional, any vNic profile that the mapping isn’t
+ supplied for it, would be mapped to a profile with the same name, if 
+that exists, or an “empty” profile otherwise, like that’s done 
+currently.
+A missing “target_network_profile” element would mean the target profile
+ is an “empty” profile. That is useful for the case that source profile 
+is known to be used in a target system for somewhat else and an 
+alternative profile isn’t exist yet.
+
+#### GUI
+The engine would scan all VMs that are intended to be imported and will 
+provide a list of network+profile that are defined on the vNics of those
+ VMs. Then mappings could be defined (by the user) in an expandable span
+ that would be added to the import VM dialog. 
+Each line would represent a single mapping with the following fields:
+
+* Source network name
+* Target network name
+* Source vNic profile name
+* Target vNic profile name
+
+![Network mapping closed](import_vm-network_mapping_close.png)
+![Network mapping opened](import_vm-network_mapping_open.png)
+
+Initially all vNic profiles would be mapped to a matching (by name) 
+network+profile on the target or would be an “Empty” profile if no match
+ exists. Then the user would be able to change that to any desired 
+profile. The list of the available networks would be filtered according 
+to the chosen cluster and the list of the profiles will be filtered by 
+the chosen network.
+
+
+### Re-assign MAC addresses for the vNics of an imported VM
+The user will be able to request to re-assigning a new MAC instead of 
+one that is problematic in the context of the target cluster.
+The reasons for a MAC to be problematic are:
+
+* Collision - the MAC is owned by an existing vNic.
+* The MAC is out of range of the destination cluster.
+
+Note that a re-assign request modifies only problematic MACs. If a VM 
+has “good” MACs as well, they would remain.
+
+#### REST API
+The import VM request will have additional boolean 
+parameter “reassign_bad_macs”. In order to keep backward compatibility 
+the default value of the new parameter will be False.
+
+```
+POST /storagedomains/{storagedomain:id}/vms/{vm:id}/import
+```
+
+```xml
+<action>
+    <cluster id=”XXX”/>
+    <storage_domain id=”YYY”/>
+    ....
+    <!-- The new addition start -->
+    <reassign_bad_macs>true<reassign_bad_macs>
+    <!-- The new addition end -->
+</action>
+```
+
+#### GUI
+VM with a problematic MAC would be marked by a warning triangle with an 
+explanatory tooltip. Checking re-assign for that VM may eliminate the 
+warning (if the warning is only due to MAC problem).
+![](register_vm-reallocate_mac.png)
+
+## Related RFEs
+* [BZ#1277675](https://bugzilla.redhat.com/show_bug.cgi?id=1277675)
+* [BZ#1317447](https://bugzilla.redhat.com/show_bug.cgi?id=1317447)
+
+## Future plans
+Apply the similar changes to import VM from the other sources flows
+ (e.g. export domain, V2V).
+ 
+## Open Issues
```

Comment:
* import a VM with a network that does not exist on destination DC, should become Empty (un-wired)
* import a VM with existing network ends with the VM properly connected
* verify that network and profile can be mapped, and that the changes apply to imported VM
* verify that reassigning MACs actually works, and that good MACs are given.
* verify that if we import a VM with a colliding MAC we actually see a warning sign.